src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -249,14 +249,14 @@ abstract class ManagedStrategy(
             // If strategy has not detected a sudden invocation result,
             // then return, otherwise process the sudden result.
             val suddenResult = suddenInvocationResult ?: return result
-            // Unexpected `ThreadAbortedError` should be thrown.
-            check(result is UnexpectedExceptionInvocationResult)
             // Check if an invocation replay is required
             val isReplayRequired = (suddenResult is SpinCycleFoundAndReplayRequired)
             if (isReplayRequired) {
                 enableSpinCycleReplay()
                 continue
             }
+            // Unexpected `ThreadAbortedError` should be thrown.
+            check(result is UnexpectedExceptionInvocationResult)
```

Comment:
Is transferring this `check` below a necessary change?

Reply:
Yes it was on purpose. Looks like there could be cases when `suddenResult is SpinCycleFoundAndReplayRequired`, but `result !is UnexpectedExceptionInvocationResult`.

Reply:
Could such case be added as a test then?